### PR TITLE
[Iga] Shell 3p KL - python test

### PR DESCRIPTION
**Pull request by tteschemacher**
**Description**
This PR adds a python-based shell 3p test to the iga application, which as far as I remember originally comes from @oberbichler .

This test has been tested already within #6608. However, to have a clean history I added it here within a new PR.

### CODE REVIEW

**Comment by tteschemacher** [comment]
@rickyaristio it seems like the enumerate isn't working.

Anyways, many thanks for checking it, and also the other changes!

Could you please approve if you agree on the test? ;)

**Comment by rickyaristio** [comment]
> @rickyaristio it seems like the enumerate isn't working.
> 
> Anyways, many thanks for checking it, and also the other changes!
> 
> Could you please approve if you agree on the test? ;)

Yes, it seems we could not use the enumerate for this case. 
I don't have any more comments and many thanks for the test implementation.

**Review by rickyaristio — APPROVED** [review]

**Comment by tteschemacher** [comment]
Thanks!

**Comment by oberbichler** [comment]
Something like

```python
for i, quadrature_point_geometry in enumerate(quadrature_point_geometries):
    model_part.CreateNewElement('Shell3pElement', element_id, quadrature_point_geometry, shell_properties)
```

might work. But that is a matter of taste.

I can remember the test. Meanwhile I tend to check the load/stiffness directly. But the test is still useful to check the workflow👍

File: applications/IgaApplication/tests/shell_3p_element_tests.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        # create a nurbs surface
+        surface = create_geometry(model_part)
+
+        # create quadrature_point_geometries
+        quadrature_point_geometries = KM.GeometriesVector()
+
+        surface.CreateQuadraturePointGeometries(quadrature_point_geometries, 3)
+
+        element_id = 1
+        for i in range(0, len(quadrature_point_geometries)):
```
Comment (oberbichler) [inline applications/IgaApplication/tests/shell_3p_element_tests.py:46]: I am not familiar with the Kratos vector but would

```python
for quadrature_point_geometry in quadrature_point_geometries:
    model_part.CreateNewElement('Shell3pElement', element_id, quadrature_point_geometry, shell_properties)
```

work?

Comment (tteschemacher) [reply applications/IgaApplication/tests/shell_3p_element_tests.py:46]: Unfortunately, that isn't working. I've even explicitly checked it. 

The problem is that the vector counter-part is a quite weird vector with an access combination of member and pointer. Probably, one could check the correct export of the vector, but I want to keep that for future research ;).

Thanks for suggesting it, though.

**Comment by tteschemacher** [comment]
> Something like
> 
> ```python
> for i, quadrature_point_geometry in enumerate(quadrature_point_geometries):
>     model_part.CreateNewElement('Shell3pElement', element_id, quadrature_point_geometry, shell_properties)
> ```
> 
> might work. But that is a matter of taste.
Same for the other suggestion, I tried it but it doesn't work in this specific vector case.

> 
> I can remember the test. Meanwhile I tend to check the load/stiffness directly. But the test is still useful to check the workflow👍 

Yes exactly. And it's quite nice as a suggestion to implement a fully python based analysis.

Thanks for your review!

**Pull request merged**